resources/persist/src/lib.rs

```diff
@@ -111,13 +158,95 @@ mod tests {
         assert_eq!(result, "test");
     }
 
+    #[test]
+    fn test_list_and_size() {
+        let persist = PersistInstance {
+            service_name: ServiceName::from_str("test1").unwrap(),
```

Comment:
The tests are looking really good now! However, there is one case that is not yet covered, but it might simplify the code.

If list/load/remove etc are called before any key has been saved, they will fail due to the folder not existing yet. This can be avoided if we ensure the folder exists before any method is called.

My idea is to add a method `new(service_name: ServiceName) -> Result<Self, PersistError>` that creates the folder upfront. Creating the folder at this time also means it is not needed in `save`.
This would mean all constructions of `PersistInstance { ... }` change to `PersistInstance::new(...)` + `.unwrap()`/`?`. It also means that the error needs to `map_err` into a `shuttle_service::Error` (or other way of converting) in `ResourceBuilder::output`.

Reply:
So a new method after all then...got it.

Reply:
I'm doing a push as a checkpoint to show where I'm at.  As a start, I've made a skeleton new method that instantiates a PersistInstance struct, given a ServiceName. I've substituted this in the tests and all works as before. I'm having difficulty now understanding how to get the folder creation incorporated.  If I put a &self in as a parameter to new, how do I then instantiate in the tests?

Reply:
> It also means that the error needs to `map_err` into a `shuttle_service::Error` (or other way of converting) in `ResourceBuilder::output`.

Having some difficulty now with this piece.  I understand what you're saying but am not sure how to make it happen. I will reflect through the day today.

Reply:
Alright, time to admit it.  After sititng staring at the code for a fair bit of time, I have no clue how to convert the errors into a shuttle_service::Error in ResourceBuilder::output.  I think I need a hint.

Reply:
This works
```rs
// Get the PersistError in, let's say, a match statement, then
return Err(shuttle_service::Error::Custom(
    /*The PersistError*/.into(),
));
```

Reply:
Ok, I feel I've got the new method working. It will convert a PersistError into a shuttle_service::Error. I'm having difficulty with a test though. All I managed to do yesterday evening was test ServiceName, which panics if you pass it something invalid. This doesn't really test the new method erroring out because of an issue with create_dir_all.  Also, the cargo-shuttle circleci check is still failing, and I'm not clear why (I have tried to read the log output) or how to resolve it.

Reply:
Yeah, the create_dir_all is hard to test failures of. Perhaps creating a folder in a root-owned directory, but that might not be reproducible on all machines. I would say that we can trust it without a test.

The CI fail on cargo-shuttle init is a sporadic error (a hard one to fix :/ ), and not related to your changes.